### PR TITLE
Core/Creatures: Allow profession tools to be used as equipment

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**
-  Allow profession tools to be used as equipment

**Issues addressed:**
None

**Tests performed:**
Builds, tested in-game: [Lee Sunspark](https://www.wowhead.com/npc=57617/lee-sunspark)

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

File: src/server/game/Entities/Item/ItemTemplate.h [hunk 1]
```diff
@@ -415,6 +415,20 @@ enum InventoryType : uint8
 
 #define MAX_INVTYPE                               35
 
+constexpr std::array<InventoryType, 10> InventoryTypesEquipable =
```
Comment (Shauren) [inline src/server/game/Entities/Item/ItemTemplate.h:418]: Eh, making it local in ObjectMgr::LoadEquipmentTemplates seems fine (I don't see another use for this)

Comment (meji46) [reply src/server/game/Entities/Item/ItemTemplate.h]: True, I don't know why I was thinking about validations for SAI. Fixed

Comment (Shauren) [reply src/server/game/Entities/Item/ItemTemplate.h]: Maybe because SAI can set random items (not validated) if action_param1 = 0

Comment (meji46) [reply src/server/game/Entities/Item/ItemTemplate.h]: Sooo... should I move it back to where I put it? I don't like the idea of being able to try to equip any item in that SAI action

**Pull request merged**